### PR TITLE
Stats revamp v2 total comments detail screen

### LINKED ISSUE #16496: [android] Stats: Total Likes Details Screen
Contains

- Total Likes card
- Top Posts card

### LINKED ISSUE #16508: [android] Stats: Total Comments Details Screen
Contains

- Total Comments card
- Top Six Posts card
- Top Commentators card

### PR DESCRIPTION
**Pull request by ravishanker**
This PR implements **Total Comments** detail screen with **Total Comments**, **Top Commentators** and **Posts Comments** cards

- [x] To do Update [FluxC part](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2386) before merge

Fixes #16508  

To test:

Setup:

-  Go to App Settings (Tap on Avatar at the top right hand corner on My Site -> Find App Settings)
-  Select Debug Settings
-  Find StatsRevampV2FeatureConfig under Features in development enable it and restart app

Test 1:

- Launch app
- Go to stats either using quick links or menu
- Ensure that it opens in Insights tab otherwise switch to Insights tab
- Notice a new Top Comments card is shown in Insights tab (scroll down if necessary)
- If the Total Comments cards is not present, go to the stats management (either through ⚙️ menu at the top right hand corner or Add new stats card at the bottom) and add the Total Comments card and save, then return to Insights.
- Tap on VIEW_MORE on the top right hand corner on the Total Comments card
- Ensure it opens the detail screen as shown above with **Total Comments**, **Top Commentators** and **Posts Comments** cards (scroll down if necessary)

NOTE:

- This PR is only for navigation and detail screen with relevant existing cards
- Cards themselves will be updated in their own PRs
- Date selector is hidden as per discussions. Will add in a separate PR if required

## Regression Notes
1. Potential unintended areas of impact
None, as it is a new screen

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Unit testing and Manual testing

3. What automated tests I added (or what prevented me from doing so)
existing unit tests

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/AuthorsCommentsUseCase.kt [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+private const val BLOCK_ITEM_COUNT = 6
+
+class AuthorsCommentsUseCase @Inject constructor(
+    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher,
+    @Named(BG_THREAD) private val backgroundDispatcher: CoroutineDispatcher,
+    private val commentsStore: CommentsStore,
+    private val statsSiteProvider: StatsSiteProvider,
+    private val resourceProvider: ResourceProvider,
+    private val statsUtils: StatsUtils,
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/AuthorsCommentsUseCase.kt]: `resourceProvider` and `statsUtils` are not being used. If you don't have a plan to use them in the future tasks of Stats Revamp, we can remove them.

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/AuthorsCommentsUseCase.kt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+            else -> State.Empty()
+        }
+    }
+
+    override suspend fun loadCachedData(): CommentsModel? {
+        val cacheMode = LimitMode.Top(BLOCK_ITEM_COUNT)
+        return commentsStore.getComments(statsSiteProvider.siteModel, cacheMode)
+    }
+
+    override fun buildLoadingItem(): List<BlockListItem> = listOf(Title(string.stats_view_comments))
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/AuthorsCommentsUseCase.kt]: Loading title string is different from the normal title here. We can use `listOf(buildTitle())` here, and title strings would be the same.

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/PostsCommentsUseCase.kt [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    override suspend fun loadCachedData(): CommentsModel? {
+        val cacheMode = LimitMode.Top(BLOCK_ITEM_COUNT)
+        return commentsStore.getComments(statsSiteProvider.siteModel, cacheMode)
+    }
+
+    override fun buildLoadingItem(): List<BlockListItem> = listOf(Title(string.posts))
+
+    override fun buildEmptyItem(): List<BlockListItem> {
+        return listOf(buildTitle(), Empty())
+    }
```
Comment (irfano) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/PostsCommentsUseCase.kt]: Loading title and empty title are different. Did you do that on purpose?

**Review by irfano — CHANGES_REQUESTED** [review]
I added small comments. We're good to merge after you review them.

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/PostsCommentsUseCase.kt (hunk 3 above)
Comment (ravishanker) [reply WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/PostsCommentsUseCase.kt]: Nope, same copy/paste carry over I guess.  Update to use buildTitle() for consistency.  Thank you

**Review by irfano — APPROVED** [review]
LGTM! 👍🏻

**Pull request merged**